src/util/sort_key.rs

```diff
@@ -0,0 +1,85 @@
+// Rust Bitcoin Library
+// Written in 2020 by
+//     rust-bitcoin developers
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+//! Utility for key Sorting
+//!
+//! Sort `bitcoin::PublicKey` according to the logic implemented in
+//! bitcoin core. In bitcoin core, keys are merely wrappers over
+//! bytes and the comparison function directly checks the underlying
+//! bytes.
+//! When keys are compressed, this utility can be used for BIP67
+//! comparison.
+use std::cmp;
+
+use PublicKey;
+
+/// Utility structure for representing ordering of PublicKeys
+/// as implemented in bitcoin core. Compares the underlying
+/// bytes when comparing the two PublicKeys in compressed form
+/// instead of naturally derived order which compares underlying
+/// 64 byte secp key.
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub struct SortKey<'pk>(&'pk PublicKey);
+
+impl<'pk> From<&'pk PublicKey> for SortKey<'pk> {
+    fn from(pk: &'pk PublicKey) -> Self {
+        SortKey(pk)
+    }
+}
+
+impl<'pk> cmp::Ord for SortKey<'pk> {
+    fn cmp(&self, other: &Self) -> cmp::Ordering {
+        match (self.0.compressed, other.0.compressed) {
+            (true, false) => cmp::Ordering::Less,
+            (false, true) => cmp::Ordering::Greater,
```

Comment:
Any specific reason why compressed keys should go before uncompressed? I am not against; just trying to make sure this is not a random choice.

Reply:
Yes, this is trying to emulate how core compares `bitcoin::PublicKey` and has nothing to do with `bip67`. Core treats public keys are byte blobs, and compares byte by byte. So, `0x02/0x03` go before `0x04`